File: grind75/Jan14/438.go
func findAnagrams(s string, p string) []int {
    ss, pp := [26]int{}, [26]int{}
    ret := []int{}
    if len(s) < len(p) {
        return ret
    }
    for _, c := range p {
        pp[int(c-'a')]++
    }
    l,r := 0, 0
    for r < len(p) {
        ss[int(s[r]-'a')]++
        r++
    }
    if ss == pp {
        ret = append(ret, l)
    }
    for r < len(s) {
        ss[int(s[l]-'a')]--
        ss[int(s[r]-'a')]++
        l++
        r++
        if ss == pp {
            ret = append(ret, l)
        }
    }
    return ret
}

File: grind75/Jan14/438.py
from collections import Counter, defaultdict
class Solution:
    def findAnagrams(self, s: str, p: str) -> List[int]:
        if len(p) > len(s):
            return []
        target = Counter(p)
        n = len(p)
        current = defaultdict(int)
        result = []
        for i in range(n-1):
            current[s[i]] += 1
        i = 0
        while i <= len(s) - n:
            current[s[i+n-1]] += 1
            if current == target:
                result.append(i)
            current[s[i]] -= 1
            if current[s[i]] == 0:
                del current[s[i]]
            i += 1
        return result
